web/themes/joinup/joinup_theme.theme

```diff
@@ -89,11 +103,27 @@ function joinup_theme_preprocess_page(&$variables) {
 
     if (in_array($rdf_type, ['collection', 'solution'])) {
       $variables['has_vertical_tabs'] = TRUE;
+      $is_form_container = TRUE;
+    }
+  }
+
+  // Check pattern for add pages.
+  $form_pattern = '.add';
```

Comment:
I debugged a bit, and even then I couldn't really understand the checks.
So, yes, please use a better strategy for checking routes.

Also, as mentioned before, these 2 checks are very similar and probably you'll manage to combine them in cascade:

```php
  // Check pattern for add pages.
  $form_pattern = '.add';
  $route_length = strlen($route_name);
  $pattern_length = strlen($form_pattern);
  if ($route_length > $pattern_length) {
    if (substr_compare($route_name, $form_pattern, $route_length - $pattern_length, $pattern_length) === 0) {
      $variables['is_blue_header'] = TRUE;
      $is_form_container = TRUE;
    }
  }

  if (!empty($route_name) && in_array($route_name, $blue_header_routes)) {
    $variables['is_blue_header'] = TRUE;
    $is_form_container = TRUE;
  }
```

Also, the iffing is repeating in other part of the code, you can abstract to a function taking the route as a parameter and telling you whether or not this page is special.